src/miral/basic_window_manager.cpp

```diff
@@ -89,25 +91,53 @@ miral::BasicWindowManager::Locker::Locker(BasicWindowManager* self) :
         self->workspaces_to_windows.left.erase(workspace);
 }
 
+miral::BasicWindowManager::LockListener::LockListener(
+    std::function<void()> const& on_lock_cb,
+    std::function<void()> const& on_unlock_cb)
+    : on_lock_cb{on_lock_cb},
+      on_unlock_cb{on_unlock_cb}
+{
+}
+
+void miral::BasicWindowManager::LockListener::on_lock()
+{
+    on_lock_cb();
+}
+
+void miral::BasicWindowManager::LockListener::on_unlock()
+{
+    on_unlock_cb();
+}
+
 miral::BasicWindowManager::BasicWindowManager(
     shell::FocusController* focus_controller,
     std::shared_ptr<shell::DisplayLayout> const& display_layout,
     std::shared_ptr<mir::shell::PersistentSurfaceStore> const& persistent_surface_store,
     mir::ObserverRegistrar<mir::graphics::DisplayConfigurationObserver>& display_configuration_observers,
-    WindowManagementPolicyBuilder const& build) :
+    WindowManagementPolicyBuilder const& build,
+    std::shared_ptr<mir::frontend::SurfaceStack> const& surface_stack,
```

Comment:
`miral` code shouldn't be referencing types from `frontend`, that leads to unnecessary coupling

Reply:
_Maybe_ we don't need the locking idea in the `BasicWindowManager` at all. See below

Reply:
Either way, this needs addressing